services/wallet/responses/router_transactions.go

```diff
@@ -0,0 +1,72 @@
+package responses
+
+import (
+	"github.com/status-im/status-go/errors"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/transactions"
+)
+
+type SendDetails struct {
+	Uuid                string                `json:"uuid"`
+	SendType            int                   `json:"sendType"`
+	FromAddress         types.Address         `json:"fromAddress"`
+	ToAddress           types.Address         `json:"toAddress"`
+	FromToken           string                `json:"fromToken"`
+	ToToken             string                `json:"toToken"`
+	FromAmount          string                `json:"fromAmount"` // total amount
+	ToAmount            string                `json:"toAmount"`
+	OwnerTokenBeingSent bool                  `json:"ownerTokenBeingSent"`
+	ErrorResponse       *errors.ErrorResponse `json:"errorResponse,omitempty"`
+
+	Username  string `json:"username"`
+	PublicKey string `json:"publicKey"`
+	PackID    string `json:"packId"`
+}
+
+type SigningDetails struct {
+	Address       types.Address `json:"address"`
+	AddressPath   string        `json:"addressPath"`
+	KeyUid        string        `json:"keyUid"`
+	SignOnKeycard bool          `json:"signOnKeycard"`
+	Hashes        []types.Hash  `json:"hashes"`
+}
+
+type RouterTransactionsForSigning struct {
+	SendDetails    *SendDetails    `json:"sendDetails"`
+	SigningDetails *SigningDetails `json:"signingDetails"`
+}
+
+type RouterSentTransaction struct {
+	FromAddress types.Address `json:"fromAddress"`
+	ToAddress   types.Address `json:"toAddress"`
+	FromChain   uint64        `json:"fromChain"`
+	ToChain     uint64        `json:"toChain"`
+	FromToken   string        `json:"fromToken"`
+	ToToken     string        `json:"toToken"`
+	Amount      string        `json:"amount"` // amount of the transaction
```

Comment:
shouldn't we have a From/To amount here as well?

Reply:
If there is a need we can add it later, but for now we don't need it.
There is `RouterSentTransactions` type that is returned.
```
type RouterSentTransactions struct {
	SendDetails      *SendDetails             `json:"sendDetails"`
	SentTransactions []*RouterSentTransaction `json:"sentTransactions"`
}
```
so it contains send details where we already return "from" and "to" amounts and every tx from the `SendTransactions` list contains an amount that was used in that tx. So far only swap can have "to" amount and since every response contains send details + sent transactions, it's easy to deduce if an amount in tx was used as "from" or "to".